repair/repair.cc

```diff
@@ -756,9 +756,14 @@ future<> repair_task_impl::do_repair_ranges(lw_shared_ptr<repair_info> ri) {
         // repair all the ranges in limited parallelism
         rlogger.info("repair[{}]: Started to repair {} out of {} tables in keyspace={}, table={}, table_id={}, repair_reason={}",
                 ri->id.uuid(), idx + 1, ri->table_ids.size(), ri->keyspace, table_name, table_id, ri->reason);
-        co_await coroutine::parallel_for_each(ri->ranges, [ri, table_id] (auto&& range) {
-            return with_semaphore(ri->rs.get_repair_module().range_parallelism_semaphore(), 1, [ri, &range, table_id] {
-                return ri->repair_range(range, table_id).then([ri] {
+        co_await coroutine::parallel_for_each(ri->ranges, [ri, table_id, parent_data] (auto&& range) -> future<> {
+            auto tmp_ri = ri;
+            auto task_impl_ptr = std::make_unique<range_repair_task_impl>(ri->rs.get_repair_module().shared_from_this(), tasks::task_id::create_random_id(), 0, ri->keyspace, "", format("{}", ri->reason), "", parent_data.id, ri, range, table_id);
+            auto task = co_await ri->rs.get_repair_module().make_task(std::move(task_impl_ptr), parent_data);
```

Comment:
```
    range_repair_task_impl(tasks::task_manager::module_ptr module, tasks::task_id id, unsigned sequence_number, std::string keyspace, std::string table, std::string type, std::string entity, tasks::task_id parent_id, lw_shared_ptr<repair_info> ri, dht::token_range range, table_id table_id)
        : repair_task_impl(module, id, sequence_number, std::move(keyspace), std::move(table), std::move(type), std::move(entity), parent_id)
        , _ri(ri)
        , _range(range)
        , _table_id(table_id)
    {}

    virtual future<tasks::task_manager::task::progress> get_progress() const override {
        auto state = _status.state;
        co_return tasks::task_manager::task::progress{
            .completed = state == tasks::task_manager::task_state::done || state == tasks::task_manager::task_state::failed,
            .total = 1.0
        };
    }
```

```
        co_await coroutine::parallel_for_each(ri->ranges, [ri, table_id, parent_data] (auto&& range) -> future<> {
            auto tmp_ri = ri;
            auto task_impl_ptr = std::make_unique<range_repair_task_impl>(ri->rs.get_repair_module().shared_from_this(), tasks::task_id::create_random_id(), 0, ri->keyspace, "", format("{}", ri->reason), "", parent_data.id, ri, range, table_id);
```

I think the way to calculate  the progress is not correct here.

so, each range_repair_task_impl crreated contribute 1 to total range. The correctness of the progress depends on if the 

co_await coroutine::parallel_for_each has finished creating all the task ranges. 

Also, we have

```
future<> repair_task_impl::do_repair_ranges(lw_shared_ptr<repair_info> ri) {
    auto parent_data = get_repair_uniq_id().task_data;
    // Repair tables in the keyspace one after another
    assert(ri->table_names().size() == ri->table_ids.size());
    for (int idx = 0; idx < ri->table_ids.size(); idx++) {
        auto table_id = ri->table_ids[idx];
        auto table_name = ri->table_names()[idx];

```

we process the ranges table by table, so the "total"  is always wrong until the last table is processed.